samtranslator/model/eventsources/pull.py

```diff
@@ -287,7 +287,26 @@ def get_policy_arn(self):  # type: ignore[no-untyped-def]
         return ArnGenerator.generate_aws_managed_policy_arn("service-role/AWSLambdaMSKExecutionRole")  # type: ignore[no-untyped-call]
 
     def get_policy_statements(self):  # type: ignore[no-untyped-def]
```

Comment:
I see this changes `MSK` events; which part adds in this PR adds support for `SelfManagedKafka`?

Reply:
SelfManagedKafka is the last class in the file.

Reply:
Ah I see, it adds a new `AUTH_MECHANISM`, and the code checks whether `Type` is one of them. Then the question becomes why is `MSK` different? Asked in another comment.